### ISSUE TITLE
Support end to end encryption

### ISSUE THREAD

**Issue by mvgorcum** [tijder/SmsMatrix#26]
If you want to use this app on a public matrix server, the admin of that server will technically have access to the plain text of all the messages sent through the app and this could be a serious leak of privacy.
If you host your own server that doesn't have to be a big deal, which is my use case, but for more general use end to end encryption would be a good solution to this problem.

Since you already use matrix-android-sdk it should be possible to support e2ee.

**Comment by tijder** [comment]
Good point to add support for e2ee. I will look to this when I have time. 
When I started building this app I got an error with the sdk when saving the data to the db. I need to fix this before using e2ee otherwise the sms client will use a different key every time.

**Comment by MurzNN** [comment]
Can you link to issue with this problem, maybe it already fixed?

**Comment by 1anomaly1** [comment]
WHY EVEN WAKE UP IN THE MORNING WITHOUT ENCRYPTION. "WHEN YOU HAVE TIME"? FOR CRYING OUT LOUD!! YOUR A DISGRACE TO THE PRIVACY COMMUNITY. GET THE HELL OUT OF MY SIGHT. YOUR FIRED!!! AND YOUR LITTLE WANNABES ON THIS PAGE ARE FIRED!!! IF P2P ISN'T THE FIRST THING, AFTER GOD, THAT YOU THINK OF IN THE MORNING, THEN STOP BREATHING OUR PRECIOUS AIR. DO YOU EVER ASK YOURSELF WHY YOU EVEN EXIST, OTHER THAN TO WASTE, AND DISTRACT FROM THE FIGHT? LEAD, FOLLOW, OR GET THE HELL OUT OF THE WAY FAGGOT. SAME GOES FOR ANYBODY WHO DISAGREES WITH MY UTTER DISDAIN FOR THIS COMPLETE WASTE OF SPACE, AND TIME.

**Comment by Pneumaticat** [comment]
@1anomaly1 no u

**Comment by freijon** [comment]
What is the progress on this after two years?
My homeserver admin recently enabled "forced encryption" on Synapse for 1:1 chats, which makes SmsMatrix unusable for me :(

**Comment by trymeouteh** [comment]
+1 for E2EE

**Comment by asakura42** [comment]
+1

**Comment by golompiki** [comment]
Really, really needed. If encryption is enabled, this would be the first good, easy and private sms forwarding service on the internets. Please, please, please

**Comment by nose-gnome** [comment]
+1
